### PR TITLE
Improve errors in common manifest API mistakes

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

File: Sources/PackageDescription4/PackageDependency.swift [hunk 1]
```diff
@@ -60,3 +60,26 @@ extension Package.Dependency: Equatable {
...
+
+// Mark common APIs used by mistake as unavailable to provide better error messages.
+extension Package.Dependency {
+    @available(*, unavailable, message: "use package(url:_:) with the .exact(Version) initializer instead")
+    public static func package(url: String, version: Version) -> Package.Dependency {
+        fatalError()
+    }
+
+    @available(*, unavailable, message: "use package(url:_:) with the .branch(String) initializer instead")
+    public static func package(url: String, branch: String) -> Package.Dependency {
```
Comment (aciidb0mb3r) [inline Sources/PackageDescription4/PackageDependency.swift]: LGTM but we should get at least one test case (see PD4LoadingTests.swift)

**Comment by hartbit** [comment]
I added the necessary test.

File: Tests/PackageLoadingTests/PD4LoadingTests.swift [hunk 2]
```diff
@@ -300,6 +308,30 @@ class PackageDescription4LoadingTests: XCTestCase {
...
+        stream <<< "       .package(url: \"/foo1\", version: \"1.0.0\")," <<< "\n"
+        stream <<< "       .package(url: \"/foo2\", branch: \"master\")," <<< "\n"
+        stream <<< "       .package(url: \"/foo3\", revision: \"rev\")," <<< "\n"
+        stream <<< "       .package(url: \"/foo4\", range: \"1.0.0\"..<\"1.5.0\")," <<< "\n"
+        stream <<< "   ]" <<< "\n"
+        stream <<< ")" <<< "\n"
+        do {
+            try loadManifestThrowing(stream.bytes) { manifest in
+                XCTFail("this package should not load succesfully")
+            }
```
Comment (aciidb0mb3r) [inline Tests/PackageLoadingTests/PD4LoadingTests.swift]: We should put a `XCTFail` here as well.

File: Tests/PackageLoadingTests/PD4LoadingTests.swift [hunk 3]
```diff
@@ -300,6 +308,30 @@ class PackageDescription4LoadingTests: XCTestCase {
...
+        stream <<< "import PackageDescription" <<< "\n"
+        stream <<< "let package = Package(" <<< "\n"
+        stream <<< "   name: \"Foo\"," <<< "\n"
+        stream <<< "   dependencies: [" <<< "\n"
+        stream <<< "       .package(url: \"/foo1\", version: \"1.0.0\")," <<< "\n"
+        stream <<< "       .package(url: \"/foo2\", branch: \"master\")," <<< "\n"
+        stream <<< "       .package(url: \"/foo3\", revision: \"rev\")," <<< "\n"
+        stream <<< "       .package(url: \"/foo4\", range: \"1.0.0\"..<\"1.5.0\")," <<< "\n"
+        stream <<< "   ]" <<< "\n"
+        stream <<< ")" <<< "\n"
```
Comment (aciidb0mb3r) [inline Tests/PackageLoadingTests/PD4LoadingTests.swift]: nit: can also use multiline string now \o/

File: Tests/PackageLoadingTests/PD4LoadingTests.swift (hunk 2 above)
Comment (hartbit) [inline Tests/PackageLoadingTests/PD4LoadingTests.swift]: Where? What for?

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**